tests/test_zarr_encryption_ipfs.py

```
@@ -0,0 +1,159 @@
+    assert np.array_equal(loaded_ds1["precip"].values, expected_ds["precip"].values), (
+        "Precip values in loaded_ds1 and expected_ds are not identical!"
+    )
+
+    # Create new encryption filter but with a different encryption key
+    encryption_key = get_random_bytes(32).hex()
+    EncryptionCodec.set_encryption_key(encryption_key)
+    register_codec(EncryptionCodec(header="dClimate-Zarr"))
+
+    loaded_failure = xr.open_zarr(store=hamt1_read)
+    # Accessing data should raise an exception since we don't have the correct encryption key
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)